### ISSUE TITLE
plot-pict does not return a typed/pict type

### ISSUE THREAD

**Issue by bennn** [racket/plot#15]
On Racket v6.4.0.15, this code fails to compile. Casting `p` to a `pict` type makes the program compile and run, but the cast shouldn't be necessary.

```
#lang typed/racket/base

(require plot/typed/no-gui
         typed/pict)

(define p (plot-pict '() #:x-min 0 #:x-max 0 #:y-min 0 #:y-max 0))
(hc-append 0
           p
           (blank 0 0))
```

The error message

```
Type Checker: type mismatch
  expected: pict
  given: Pict
  in: p
```

**Comment by alex-hhh** [comment]
Hi @samth , I might need your advice on this issue.  This is still a problem in Racket 7.7.  The type `Pict` is defined in https://github.com/racket/plot/blob/6a593ce8e393221e109335d56e9fb65ddc72bfd3/plot-lib/plot/private/no-gui/evil-types.rkt#L7

Was this done before the `pict` library was typed?  Is this still needed?

**Comment by samth** [comment]
Yes, I believe that code predates `typed/pict`. I'm sure I have some code lying around that fixes this, but I think it's not backwards compatible in other ways, which is why it didn't get merged. Unfortunately, I can't find it at the moment. I think the short answer is that this is likely to stay a bug for a while yet unless it's really a blocker for someone.

**Comment by bdeket** [comment]
Related to the original question, and not what @gpitton reported:
In Pr #90 I edited the `evil-typed.rkt` to re-export `pict` from `typed/pict` as `Pict` instead of creating the opaque type.
I hope I didn't break anything doing that. But with this change, the original example works.

**Comment by samth** [comment]
@gpitton That isn't a bug, the problem is that mutable vectors are not subtypes in the way you are expecting in that code. You can't supply a `(Mutable-Vector Flonum Integer)` when you need a  `(Mutable-Vector Any (U False Real ivl))`. That's because `discrete-histogram` could (but doesn't) mutate the input to have `'x` as the 0th element of the inner vector, violating the type of `bins`.

Depending on how you're constructing `bins`, you might be able to use `vector-immutable` to just have immutable vectors and then it will work.

**Comment by gpitton** [comment]
> Depending on how you're constructing `bins`, you might be able to use `vector-immutable` to just have immutable vectors and then it will work.

@samth Thanks! I am not sure I understand why the type checking failed, but this is how I was generating the vector:
```racket
(define bins
  (for/vector : (Vectorof (Vector Float Integer))
      ([lb (in-list bounds)] ;; lb is a Float
       [c (in-list rw-counts)])  ;; c is an Integer
      (vector (f lb) c)))  ;; f is a function Float -> Float
```
and this is how I fixed it:
```racket
(define bins
  (for/vector : (Vectorof (Immutable-Vector Float Integer))
      ([lb (in-list bounds)]
       [c (in-list rw-counts)])
      (vector-immutable (f lb) c)))
```

As expected, the new type signature is:
```
> bins
- : (Vectorof (Immutable-Vector Flonum Integer))
```
Now everything works fine, so I take back my previous report.

**Comment by alex-hhh** [comment]
This was fixed by #90, and this change will make it in Racket 8.1.  All packages on pkgs.racket-lang.org were already built with this change and I am not aware of any regressions.   I will close this issue, since ultimately this is the intended behavior.
